pkg/build/api/v1beta3/types.go

```diff
@@ -382,6 +386,21 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// BuildPostCommitSpec holds a build post commit hook specification.
+type BuildPostCommitSpec struct {
+	// Shell is an action that executes a shell script in a container
+	// running the build output image.
+	Shell string `json:"shell,omitempty" description:"shell script to be executed in a container running the build output image"`
```

Comment:
@smarterclayton assuming that most images sets entrypoint to shell, I think @rhcarvalho is right. If user want to preserve context of the program or have quoting problem, then they can simply set `args` and get same behavior as when setting `shell`. In case you want really execute the script in shell, then having []string here makes your life harder :-) Also I think we can call the field `Script` or `ShellScript`.